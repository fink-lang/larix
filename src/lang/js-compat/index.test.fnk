{describe, it, expect, to_match_snapshot} = import '@fink/jest'

{parse_expr, parse} = import '../..'


describe 'js-compat operators', fn:

  it 'parses new', fn:
    expect
      parse_expr 'new_foo = new Foobar(spam, ni)'
      to_match_snapshot

    expect
      parse_expr 'new_foo = new Foobar spam, ni'
      to_match_snapshot


  it 'parses throw', fn:
    expect
      parse_expr `throw_foo = foo or throw err('foobar')`
      to_match_snapshot

    expect
      parse_expr `throw_foo = foo or throw err 'foobar'`
      to_match_snapshot


describe 'try ...:', fn:

  it 'parses', fn:
    expect
      parse `
        [err, result] = try:
          throw new Error 'foobar'
      `
      to_match_snapshot



describe 'constants', fn:

  it 'parses null', fn:
    expect
      parse `
        foo = null
        bar = nullish
      `
      to_match_snapshot
