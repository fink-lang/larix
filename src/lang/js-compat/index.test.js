import {parse_expr} from '../..';


describe('js-compat operators', ()=> {
  it('parses new', ()=> {
    expect(
      parse_expr('new Foobar(spam, ni)')
    ).toMatchSnapshot();
  });

  it('parses throw', ()=> {
    expect(
      parse_expr('throw err(`foobar`)')
    ).toMatchSnapshot();
  });
});
