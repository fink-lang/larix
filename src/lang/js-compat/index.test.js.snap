// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`js-compat operators parses new 1`] = `
new (1:0-1:20)
  call (1:4-1:20)
    ident (1:4-1:10) Foobar
    ident (1:11-1:15) spam
    ident (1:17-1:19) ni

`;

exports[`js-compat operators parses throw 1`] = `
throw (1:0-1:19)
  call (1:6-1:19)
    ident (1:6-1:9) err
    string \` (1:10-1:18)
      \`foobar\`

`;
